Re-ordering of slides and templated makefile

derekpierre opened the pull request:
* Re-ordered slides for a more storied form of our slides i.e. Problem -> Technology -> Solution ...
* Minor text edits to various slides
* The slides are probably more than needed for any one event, but the presenter can decide what slides to remove based on the event.
* Made generation of event-specific slides templated through the makefile. Parameters can be passed when running make to generate a specific pdf file for an event. Non-event specific files can be performed with a BUILD=template option when making changes to master or enterprise
* Added a README to explain how to build slides.
* Branches should still be made for specific event slides
* Makefile is not my strong suit so any improvements to the makefile methodology are welcomed

derekpierre committed:
Modify makefile to programmatically set variables to use when generating slides from the template

derekpierre committed:
Added the ability for templated builds to be easily performed for non-event specific slide updates. Added a README

derekpierre committed:
Initial gamut of ordered slides

derekpierre committed:
Fix makefile

derekpierre committed:
Modified Use cases slides and removed excess svg images

derekpierre committed:
Fix network images for data sharing slides

derekpierre committed:
Improve Usage parameter names, and update README

derekpierre committed:
Add note to README

derekpierre committed:
Cleanup makefile

mswilkison commented on slides.tex:
```diff
@@ -99,103 +72,134 @@
-            \item Splitting trust between re-encryption nodes (more tokens = more trust and more work);
-            \item Proof of Stake for minting new coins according to the mining schedule;
-            \item Security deposit to be at stake against malicious behavior of nodes
+            \item Splitting trust across re-encryption nodes: more tokens = more trust and more work
```
more trust, more work, and more compensation

mswilkison commented on slides.tex:
```diff
@@ -99,103 +72,134 @@
+        \end{figure}
+        \begin{itemize}
+            \item Single node collusion with receiver possible
+            \item Single node can deny to do work.
```
extraneous '.'

mswilkison commented on slides.tex:
```diff
@@ -99,103 +72,134 @@
+            \end{itemize}
+            \item Smart contract (public) method
         \end{itemize}
+        Decentralized re-encryption nodes (Ursulas) are trusted to apply conditions without decrypting data
```
are relied on

mswilkison replied:
without having the ability to decrypt data

mswilkison commented on slides.tex:
```diff
@@ -99,103 +72,134 @@
         \begin{figure}
             \centering
-            \includegraphics[height=5.5cm]{pdf/daily-compensation.pdf}
+            \includegraphics[height=7cm]{pdf/file-sharing-alternative.pdf}
```
the arrow at the bottom is messed up for me

derekpierre replied:
Let me know if this is still a problem.

mswilkison replied:
Fixed!

mswilkison commented on slides.tex:
```diff
@@ -99,103 +72,134 @@
+
+    \begin{frame}
+        \frametitle{Use Cases}
+        \framesubtitle{Decentralized Access-Controlled Content}
```
Other subheaders aren't capitalized

mswilkison commented on slides.tex:
```diff
@@ -206,13 +210,42 @@
+       \begin{itemize}
+           \item Slow Peformance
+           \begin{itemize}
+               \item NuCypher is investing efforts in this area
```
I would just mention nuFHE explicitly

mswilkison commented on slides.tex:
```diff
@@ -240,9 +273,68 @@
+    \end{frame}
+
+    \begin{frame}
+      \frametitle{Appendix: Team}
```
Can we add karce

mswilkison commented on slides.tex:
```diff
@@ -221,7 +254,7 @@
 
     \begin{frame}
-        \frametitle{Useful links}
+        \frametitle{More Information}
```
Let's think some about what urls we want to include on this page. How about replacing 'PyUmbral' with 'Umbral reference implementation' (not sure about this). Remove goUmbral (since we'll consistently be adding more languages I don't know if it's makes sense to list them all. Add nuFHE.  Add staking economics paper? (or perhaps link to it elsewhere in the presentation)

derekpierre replied:
Added economics paper to NU Token slides
Added Umbral doc and nuFHE URLs
Updated pyUmbral url label

mswilkison replied:
Great. IMO replace umbral-doc with nucypher repo since we almost have that ready for use.

mswilkison reviewed (commented):
We should add a slide about Sputnik

derekpierre committed:
* Added Karce to Employees slide
* Added Sputnik slide
* Fixed wording on Competing Technology slide
* Removed numbers from Token Metrics
* Added link to mining & staking paper to NU Token Metrics slide
* Cleaned up links on More Information slide

derekpierre committed:
* Replace umbral-doc URL with network repos URL
* Remove absolute paths from makefile

mswilkison commented:
Looks good to me.

michwill commented:
That's a cool way to buidl it.
I wonder if even a better way to be generating a `template.tex` where all of this is defined, and if it exists, just plain `make` makes it (also the `template.tex` can be committed to branches, forks etc).

michwill commented:
(and, perhaps, by default this `template.tex` can be in `.gitignore`)

michwill commented:
On an unrelated note, my `inkscape` screws up arrow on the colorful chat slides (maybe try saving text as curves in `svg` to prevent it from happening?)
https://imgur.com/WjVeJbr

derekpierre committed:
Update alternative images to change text objects to path

derekpierre commented:
@michwill re: template.tex

I guess you mean something like this: https://tex.stackexchange.com/questions/294117/importing-parameters-from-a-separate-file/294546#294546. Generally, we would then locally change the template.tex file for specific branch builds, but since the template.tex file would be added to .gitignore the changed file would never get submitted to git.

Hmm, this would be cleaner from a make perspective, but some questions about the handling of the template.tex file:
1. Would the template.tex file be submitted to the base branches (master, enterprise) with templated values eg. "fname", and then the file listed in the .gitignore so that it doesn't ever get replaced?
2. I don't know if the case of the 'template.tex' not existing would ever occur since if it would be needed on the base branches (#1 above) for non-event specific updates to slides for easy builds.
3. For this method, I guess we can no longer enforce that the variables have been explicitly set by whoever is generating the slides. The person generating the slides would update the template.tex file locally, and hopefully replace all existing templated values but is not forced to i.e. templated values can mistakenly remain?

derekpierre commented:
@michwill modified the text in the alternative image files, is the problem fixed?

michwill commented:
@derekpierre The top recipient in a couple of files who have him has a problem (two text labels which are slightly shifted) https://imgur.com/a726hpH

derekpierre committed:
Fixed alternative images Recipient text that was doubled - interesting that it isn't noticeable on my Mac

michwill commented:
Ok, looks like pictures are good

michwill commented:
@derekpierre yes, I think having a `template.tex` instead of make options is all right.
What do you think, is it more convenient than specifying in make options? (for me - most definitely more convenient)

derekpierre commented:
@michwill - I think the template.tex methodology is definitely more convenient. The one drawback is that we can no longer enforce that the variables have been explicitly set but I agree that the convenience is worth it.

I'll work on updating the makefile/tex files accordingly.

michwill commented:
We can enforce variables in the template to be non-empty

derekpierre committed:
* Updated templated build methodology for slides
We can now simply run 'make', and provide input parameters from a template.tex file, which can be edited for event-specific information.

derekpierre committed:
Ignore template.tex file

derekpierre commented:
@michwill Using just .gitignore is a bit tricky given that the template.tex file has been committed.

It seems the most common strategy is to:
1. Have a base file like 'template.tex.example' that is tracked by git
2. Add 'template.tex' to .gitignore i.e. template.tex is no longer tracked
3. When building, copy the template.tex.example as template.tex, and edit template.tex for the event-specific information

See: https://www.reddit.com/r/git/comments/34a36i/prevent_changes_to_a_file/ and https://gist.github.com/tsrivishnu/a2f3adbbca9fcad5f3597af301ad1abb#useful-scenario-managing-configuraiton-files-in-git-repositories

derekpierre committed:
* Use template.tex.base file instead of template.tex for version of template file in repos
* Now template.tex.base should never be edited, but instead copied to template.tex and the relevant edits made for a specific event
* Updated README with instructions
* template.tex should be ignored by git
